PHP Distrib tests is broken

https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_distribtest/platform=linux/lastBuild/testReport/junit/(root)/tasks/distribtest_php_linux_x64_jessie/

Broken by #10427 

```
/bin/bash /tmp/pear/temp/pear-build-defaultuser7VZGCJ/grpc-1.4.0dev/libtool --mode=compile  
-Wall -Werror     -Wno-parentheses-equality -Wno-unused-value -std=c11     -fvisibility=hidden 
-DOPENSSL_NO_ASM -D_GNU_SOURCE -DWIN32_LEAN_AND_MEAN     -D_HAS_EXCEPTIONS=0 
-DNOMINMAX -I. -I/tmp/pear/temp/grpc -DPHP_ATOM_INC 
-I/tmp/pear/temp/pear-build-defaultuser7VZGCJ/grpc-1.4.0dev/include 
-I/tmp/pear/temp/pear-build-defaultuser7VZGCJ/grpc-1.4.0dev/main 
-I/tmp/pear/temp/grpc -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM 
-I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib 
-I/tmp/pear/temp/grpc/include -I/tmp/pear/temp/grpc/src/php/ext/grpc 
-I/tmp/pear/temp/grpc/third_party/boringssl/include -I/tmp/pear/temp/grpc/third_party/cares 
-I/tmp/pear/temp/grpc/third_party/cares/cares -I/tmp/pear/temp/grpc/third_party/cares/config_linux  
-DHAVE_CONFIG_H     -c /tmp/pear/temp/grpc/src/core/lib/surface/lame_client.cc 
-o src/core/lib/surface/lame_client.lo
libtool: compile: unrecognized option `-Wall'
libtool: compile: Try `libtool --help' for more information.
Makefile:510: recipe for target 'src/core/lib/surface/lame_client.lo' failed
make: *** [src/core/lib/surface/lame_client.lo] Error 1
ERROR: `make' failed
```

cc @jtattermusch

--- a/config.m4
+++ b/config.m4
@@ -13,9 +13,11 @@ if test "$PHP_GRPC" != "no"; then
 
   LIBS="-lpthread $LIBS"
 
+  CFLAGS="-Wall -Werror -Wno-parentheses-equality -Wno-unused-value -std=c11"
+  CXXFLAGS="-std=c++11"
   GRPC_SHARED_LIBADD="-lpthread $GRPC_SHARED_LIBADD"
+  PHP_REQUIRE_CXX()
   PHP_ADD_LIBRARY(pthread)
-
   PHP_ADD_LIBRARY(dl,,GRPC_SHARED_LIBADD)
   PHP_ADD_LIBRARY(dl)
 
@@ -688,9 +690,8 @@ if test "$PHP_GRPC" != "no"; then
     third_party/cares/cares/inet_net_pton.c \
     third_party/cares/cares/inet_ntop.c \
     third_party/cares/cares/windows_port.c \
-    , $ext_shared, , -Wall -Werror \
-    -Wno-parentheses-equality -Wno-unused-value -std=c11 \
-    -fvisibility=hidden -DOPENSSL_NO_ASM -D_GNU_SOURCE -DWIN32_LEAN_AND_MEAN \
+    , $ext_shared, , -fvisibility=hidden \
+    -DOPENSSL_NO_ASM -D_GNU_SOURCE -DWIN32_LEAN_AND_MEAN \
     -D_HAS_EXCEPTIONS=0 -DNOMINMAX)
 
   PHP_ADD_BUILD_DIR($ext_builddir/src/php/ext/grpc)

--- a/templates/config.m4.template
+++ b/templates/config.m4.template
@@ -15,9 +15,11 @@
 
     LIBS="-lpthread $LIBS"
 
+    CFLAGS="-Wall -Werror -Wno-parentheses-equality -Wno-unused-value -std=c11"
+    CXXFLAGS="-std=c++11"
     GRPC_SHARED_LIBADD="-lpthread $GRPC_SHARED_LIBADD"
+    PHP_REQUIRE_CXX()
     PHP_ADD_LIBRARY(pthread)
-
     PHP_ADD_LIBRARY(dl,,GRPC_SHARED_LIBADD)
     PHP_ADD_LIBRARY(dl)
 
@@ -43,9 +45,8 @@
       % endfor
       % endif
       % endfor
-      , $ext_shared, , -Wall -Werror ${"\\"}
-      -Wno-parentheses-equality -Wno-unused-value -std=c11 ${"\\"}
-      -fvisibility=hidden -DOPENSSL_NO_ASM -D_GNU_SOURCE -DWIN32_LEAN_AND_MEAN ${"\\"}
+      , $ext_shared, , -fvisibility=hidden ${"\\"}
+      -DOPENSSL_NO_ASM -D_GNU_SOURCE -DWIN32_LEAN_AND_MEAN ${"\\"}
       -D_HAS_EXCEPTIONS=0 -DNOMINMAX)
 
     PHP_ADD_BUILD_DIR($ext_builddir/src/php/ext/grpc)

